lib/test/api_test.rb

```diff
@@ -0,0 +1,259 @@
+    assert_equal "Success, cstore transaction is successful", result.status_message
+    assert_equal "indomaret", result.store
+  end
+
+  def test_charge_alfamart
+    result = @mt_test.charge(
+      "payment_type": "cstore",
+      "transaction_details": {
+        "order_id": "order-alfamart-#{rand(1..10000)}",
+        "gross_amount": 44000
+      },
+      "cstore": {
+        "store": "alfamart",
+        "alfamart_free_text_1": "1st row of receipt",
+        "alfamart_free_text_2": "2nd row",
+        "alfamart_free_text_3": "3rd row"
+      }
+    )
+    assert_equal 201, result.status_code
+    assert_equal "Success, cstore transaction is successful", result.status_message
+    assert_equal "alfamart", result.store
+  end
+
+  def test_charge_gopay
+    result = @mt_test.charge(
+      "payment_type": "gopay",
+      "transaction_details": {
+        "order_id": "order-gopay-#{rand(1..10000)}",
+        "gross_amount": 44000
+      },
+      "gopay": {
+        "enable_callback": "true",
+        "callback_url": "someapps://callback"
+      }
+    )
+    assert_equal 201, result.status_code
+    assert_equal "GoPay transaction is created", result.status_message
+    assert_equal "pending", result.transaction_status
+  end
+
+  def test_charge_bcaklikpay
+    result = @mt_test.charge(
+      "payment_type": "bca_klikpay",
+      "transaction_details": {
+        "order_id": "order-bcaklikpay-#{rand(1..10000)}",
+        "gross_amount": 44000
+      },
+      "bca_klikpay": {
+        "type": "1",
+        "description": "pembelian produk"
+      }
+    )
+    assert_equal 201, result.status_code
+    assert_equal "OK, BCA KlikPay transaction is successful", result.status_message
+    assert_equal "pending", result.transaction_status
+  end
+
+  def test_charge_akulaku
+    result = @mt_test.charge(
+      "payment_type": "akulaku",
+      "transaction_details": {
+        "order_id": "order-akulaku-#{rand(1..10000)}",
+        "gross_amount": 44000
+      }
+    )
+    assert_equal 201, result.status_code
+    assert_equal "Success, Akulaku transaction is created", result.status_message
+    assert_equal "pending", result.transaction_status
+  end
+
+  def test_charge_danamon_online
+    result = @mt_test.charge(
+      "payment_type": "danamon_online",
+      "transaction_details": {
+        "order_id": "order-danamon-#{rand(1..10000)}",
+        "gross_amount": 44000
+      }
+    )
+    assert_equal "Success, Danamon Online transaction is successful", result.status_message
+    assert_equal 201, result.status_code
+    assert_equal "pending", result.transaction_status
+  end
+
+  def test_create_widget_token
+    result = @mt_test.create_widget_token(
+      transaction_details: {
+        order_id: "THE-ITEM-ORDER-#{rand(1..10000)}",
+        gross_amount: 200000
+      },
+      "credit_card": {
+        "secure": true
+      }
+    )
+    assert_equal 201, result.status_code
+  end
+
+end
```

Comment:
Any negative test case scenario? better add it, like this:
https://github.com/Midtrans/midtrans-nodejs-client/blob/master/test/coreApi.test.js#L78
search for `fail to`. Name your negative test case accordingly e.g: `test_fail_to_charge_with_empty_server_key`